org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -326,9 +327,15 @@ public static IProject createInvisibleProjectIfNotExist(IPath workspaceRoot) thr
 					workspaceLinkFolder.createLink(workspaceRoot.toFile().toURI(), IResource.REPLACE, null);
 					File settings = new File(workspaceRoot.toFile(), SETTINGS);
 					if (settings.exists()) {
+						// reset the preview feature property - https://github.com/eclipse/eclipse.jdt.ls/pull/1863#issuecomment-924395431
+						Hashtable<String, String> defaultOptions = JavaCore.getDefaultOptions();
+						String defaultPreview = defaultOptions.get(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES);
+						String defaultReport = defaultOptions.get(JavaCore.COMPILER_PB_REPORT_PREVIEW_FEATURES);
+						IJavaProject javaProject = JavaCore.create(invisibleProject);
+						javaProject.setOption(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES, defaultPreview);
+						javaProject.setOption(JavaCore.COMPILER_PB_REPORT_PREVIEW_FEATURES, defaultReport);
```

Comment:
I don't understand why is this necessary here?

Reply:
The Syntax server calls JVMConfigurator.configureJVMSettings before configuring linked settings. See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java#L319
You can try to run the InvisibleProjectImporter.testPreviewFeaturesSettingsDisabled test without it.